storage/block/block.go

```diff
@@ -365,8 +366,6 @@ func (b *dbBlock) resetSegmentWithLock(seg ts.Segment) {
 	b.retriever = nil
 	b.retrieveID = nil
 	b.wasRetrievedFromDisk = false
-
-	b.ctx.RegisterFinalizer(&seg)
```

Comment:
I see that you moved this to a synchronous call in resetRetrievableWithLock, but is that enough? Seems like it would never happen for the ResetRetrievable path

Reply:
Moved the RegisterFinalizer?  We don't use the context anymore for finalization (block no longer has a context even) on the segment so this didn't get moved anywhere per se.  Or do you mean something else?